### PR TITLE
solution

### PR DESCRIPTION
**Pull request by tykhan**
[DEMO LINK](https://tykhan.github.io/react_goods-selector/)

### CODE REVIEW

**Review by kostyasabada — APPROVED** [review]
Good!

File: src/App.js [hunk 1]
```diff
@@ -14,11 +14,56 @@ const goodsFromServer = [
...
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    status: 'Your bag is empty',
+    selected: '',
```
Comment (vkryvytskyy) [inline src/App.js:20]: probably you don't need status, remove it

File: src/App.js [hunk 2]
```diff
@@ -14,11 +14,56 @@ const goodsFromServer = [
...
+  };
+
+  render() {
+    const { status, selected } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className=" status">{status}</h1>
+        <ul className="goods">
+          {goodsFromServer.map(goods => (
```
Comment (vkryvytskyy) [inline src/App.js:37]:
```suggestion
          {goodsFromServer.map(good => (
```
this is one good

File: src/App.js [hunk 3]
```diff
@@ -14,11 +14,56 @@ const goodsFromServer = [
...
+      selected: event.target.textContent,
+    });
+  };
+
+  render() {
+    const { status, selected } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className=" status">{status}</h1>
```
Comment (vkryvytskyy) [inline src/App.js:35]:
```suggestion
        <h1 className=" status">{selected ? `${selected} is in basket` : 'empty'}</h1>
```

**Review by vkryvytskyy — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 4]
```diff
@@ -14,11 +14,56 @@ const goodsFromServer = [
...
+    status: 'Your bag is empty',
+    selected: '',
+  }
+
+  savedGoods = (event) => {
+    this.setState({
+      status: `${event.target.textContent} is in basket`,
+      selected: event.target.textContent,
+    });
+  };
```
Comment (vkryvytskyy) [inline src/App.js:28]:
```suggestion
  savedGoods = (good) => {
    this.setState({
      selected: good,
    });
  };
```

File: src/App.js [hunk 5]
```diff
@@ -14,11 +14,56 @@ const goodsFromServer = [
...
+
+    return (
+      <div className="App">
+        <h1 className=" status">{status}</h1>
+        <ul className="goods">
+          {goodsFromServer.map(goods => (
+            <li key={goods}>
+              <button
+                type="button"
+                onClick={this.savedGoods}
```
Comment (vkryvytskyy) [inline src/App.js:41]:
```suggestion
                onClick={() => this.savedGoods(good)}
```

**Review by vkryvytskyy — COMMENTED** [review]
without event.targe.textContent looks more readable, i think

File: src/App.js (hunk 2 above)
Comment (tykhan) [reply src/App.js:37]: Знімок екрана 2020-09-03 о 20 18 09

File: src/App.js [hunk 6]
```diff
@@ -14,11 +14,58 @@ const goodsFromServer = [
...
+          type="button"
+          className="remove"
+          onClick={() => {
+            this.setState({
+              selected: '',
+            });
+          }}
+        >
+          Remove
+        </button>
```
Comment (vkryvytskyy) [inline src/App.js:65]: show this button only when some goods is selected

**Review by vkryvytskyy — APPROVED** [review]

**Pull request closed without merging**